### ISSUE TITLE
Keras Code Example Category

### ISSUE THREAD

**Issue by innat** [keras-team/keras-io#2151]
### Issue Type

Feature Request

### Source

source

### Keras Version

keras 3

### Custom Code

No

### Current Behavior?

As someone who frequently mentors newcomers in machine learning, I’m often asked how to get started. One of the resources I consistently recommend is the [keras.io/code](https://keras.io/examples/) examples section. It’s a fantastic collection, well-organized by domains such as vision, NLP, reinforcement learning, and more.

However, for absolute beginners, the domain-based structure can sometimes feel overwhelming. It would be even more helpful if the examples were also grouped by experience level — for instance, **Beginner**, **Intermediate**, and **Advanced**. This kind of categorization would provide a **smoother learning path and allow learners to build confidence progressively.**

In addition, before diving into any specific code example, it would be great to include a brief section listing the prerequisite knowledge expected — such as basic Python, NumPy, understanding of the Keras Functional API, how to write a custom model or a custom training loop, and use of callbacks. These concepts are already well-documented under the [keras.io/guides](https://keras.io/guides/) section, so linking to the relevant guides from each code example would help learners better prepare and understand the material.

This is just a rough suggestion, but I believe it could make the learning experience more structured and accessible, especially for those who are just starting their ML journey. If it sounds promising, maybe it’s worth validating through a community survey — or even trying it out on a small scale. Thanks!

cc. @fchollet @mattdangerw @divyashreepathihalli 

### Standalone code to reproduce the issue or tutorial link

```shell
- https://keras.io/examples/
```

### Relevant log output

```shell

```

**Comment by sachinprasadhs** [comment]
Thank you for your interest in improving the user experience and for the valuable suggestion.

We do have multiple options at a high level to indicate good examples/guides to start.

One option is on https://keras.io/examples/, where we use a ★ mark to indicate starter examples.

We also use the Keras mascot in some guides/examples to indicate difficulty levels, and we can expand this to even more content.
<img width="1038" height="412" alt="Image" src="https://github.com/user-attachments/assets/77ae1f34-c9b1-42b1-ba71-b9ac66fd271e" />


Your suggestion to include a prerequisite block is interesting and we will discuss it within the team.

**Comment by innat** [comment]
@sachinprasadhs 
With all due respect, I think there may have been a misunderstanding. I’d kindly ask you to revisit my original message. Just to clarify — my suggestion isn't about labeling individual code examples as beginner, intermediate, or advanced (though that would certainly be helpful too). What I’m proposing is a broader categorization of the entire Keras code examples collection, adding an additional structure based on experience level across the whole section.

**Comment by JyotinderSingh** [comment]
Thank you for this suggestion! It would be valuable to incorporate these ideas into the website. While the team's capacity to work on this may be limited, we welcome contributions for the same!

**Comment by Rachit-Tiwari-7** [comment]
Hi, I'm new to the project and would like to try and tackle this issue for my first contribution. Is it still available?

**Comment by Parshv-sudo** [comment]
Hi @JyotinderSingh and @sachinprasadhs,

I’d like to start with a small, incremental contribution to keras-io by adding a short “Prerequisites” section to a few starter-friendly examples (starting with Image classification from scratch).

The section would briefly list the expected background knowledge and link to relevant existing Keras developer guides (e.g., Functional API, training with built-in methods), without introducing new difficulty labels or restructuring the examples.

For basic Python/NumPy familiarity, I plan to mention it explicitly and optionally reference the existing Quick Keras Recipes example on writing Keras models with TensorFlow NumPy, rather than linking to a non-existent developer guide.

This would serve as a pilot change that can be extended to other examples if the approach looks good.

Please let me know if this sounds reasonable.
